File: src/main/kotlin/com/github/renttrent/jetbrainsdbsecurity/listeners/SaveListener.java
package com.github.renttrent.jetbrainsdbsecurity.listeners;

import com.intellij.openapi.editor.Document;
import com.intellij.openapi.fileEditor.FileDocumentManagerListener;

import org.jetbrains.annotations.NotNull;



public final class SaveListener implements FileDocumentManagerListener {
    @Override
    public void beforeAllDocumentsSaving(){
        //println("before all documents Save");
    }

    @Override
    public void beforeDocumentSaving(@NotNull Document document) {
        //println("before document save " + document.getClass().toString());
    }

}

File: src/main/kotlin/com/github/renttrent/jetbrainsdbsecurity/listeners/SaveListener.kt
package com.github.renttrent.jetbrainsdbsecurity.listeners

import com.github.renttrent.jetbrainsdbsecurity.actions.SqlInjectionDetectionAction
import com.intellij.openapi.editor.Document
import com.intellij.openapi.fileEditor.FileDocumentManager
import com.intellij.openapi.fileEditor.FileDocumentManagerListener
import com.intellij.openapi.project.Project
import com.intellij.openapi.project.guessProjectForFile
import com.intellij.openapi.vfs.VirtualFile


class SaveListener : FileDocumentManagerListener {

    override fun beforeDocumentSaving(document: Document) {
        val file: VirtualFile? = FileDocumentManager.getInstance().getFile(document)
        val project: Project? = file?.let { guessProjectForFile(it) }

        SqlInjectionDetectionAction().parseDocument(project!!, document)
    }
}

